applications/IgaApplication/tests/shell_kl_discrete_element_test.py

```diff
@@ -0,0 +1,294 @@
+#
+#   KRATOS  _____________
+#          /  _/ ____/   |
+#          / // / __/ /| |
+#        _/ // /_/ / ___ |
+#       /___/\____/_/  |_| Application
+#
+#   Main authors:   Thomas Oberbichler
+#
+
+from KratosMultiphysics import *
+from KratosMultiphysics.IgaApplication import *
+from KratosMultiphysics.StructuralMechanicsApplication import *
+import new_linear_solver_factory
+
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+
+
+def GetFilePath(fileName):
+    return os.path.dirname(__file__) + "/" + fileName
+
+
+class ShellKLDiscreteElementTest(KratosUnittest.TestCase):
+
+    def solve_cantilever(create_geometry):
+        model_part = ModelPart('Model')
+
+        model_part.AddNodalSolutionStepVariable(DISPLACEMENT)
+        model_part.AddNodalSolutionStepVariable(REACTION)
+        model_part.AddNodalSolutionStepVariable(POINT_LOAD)
+
+        # create property for shell elements
+
+        shell_properties = Properties(1)
+        shell_properties.SetValue(THICKNESS       , 0.1      )
+        shell_properties.SetValue(YOUNG_MODULUS   , 210000000)
+        shell_properties.SetValue(POISSON_RATIO   , 0        )
+        shell_properties.SetValue(DENSITY         , 78.5     )
+        shell_properties.SetValue(CONSTITUTIVE_LAW,
+            LinearElasticPlaneStress2DLaw())
+        model_part.AddProperties(shell_properties)
```

Comment:
Usually the property is created if you do `mp.GetProperty(Id)`, then you don't have to add it (which for reasons I don't remember one should avoid)

Reply:
Then it behaves like a `std::map`?

Reply:
I only found `mp.GetProperties()[Id]`